test/simplify_qdq_test.cpp

```diff
@@ -0,0 +1,537 @@
+                                                        {"ceil_mode", 0}}),
+                                     a1);
+        auto fl = m2.add_instruction(migraphx::make_op("flatten", {{"axis", 1}}), ap);
+        auto q4 = add_quantize_op(m2, "quantizelinear", fl, l2, l1);
+        auto dot =
+            m2.add_instruction(migraphx::make_op("quant_dot", {{"alpha", 1}, {"beta", 0}}), q4, db);
+        auto d9  = add_quantize_op(m2, "dequantizelinear", dot, sc2, z2);
+        auto mb1 = m2.add_instruction(
+            migraphx::make_op("multibroadcast", {{"output_lens", {1, 1000}}}), d3);
+        auto a2 = m2.add_instruction(migraphx::make_op("add"), d9, mb1);
+        m2.add_return({a2});
+    }
+
+    run_pass(m1);
+    EXPECT(m1 == m2);
+}
+
+TEST_CASE(mobilenet_snippet)
+{
+    migraphx::shape s1{migraphx::shape::int8_type, {1}};
+    migraphx::shape s2{migraphx::shape::int8_type, {1280, 1000}};
+    migraphx::shape s3{migraphx::shape::int8_type, {1000}};
+    migraphx::shape s4{migraphx::shape::int8_type, {1280, 320, 1, 1}};
+    migraphx::shape s6{migraphx::shape::int32_type, {1280}};
+    migraphx::shape s7{migraphx::shape::float_type, {1, 320, 7, 7}};
+    migraphx::shape s8{migraphx::shape::float_type, {1}};
+
+    std::vector<float> scale{0.5};
+    std::vector<int> zero{0};
+
+    auto create_module = [&]() {
+        migraphx::module mm;
+        auto db      = mm.add_parameter("db", s2); // dot input b
+        auto ab      = mm.add_parameter("ab", s3); // add input b
+        auto weights = mm.add_parameter("weights", s4);
+        auto bias    = mm.add_parameter("bias", s6);
+        auto input   = mm.add_parameter("input", s7);
+
+        auto l1 = mm.add_literal(s1, zero);
+        auto l2 = mm.add_literal(s8, scale);
+
+        auto d1  = add_quantize_op(mm, "dequantizelinear", weights, l2, l1);
+        auto d2  = add_quantize_op(mm, "dequantizelinear", bias, l2, l1);
+        auto d3  = add_quantize_op(mm, "dequantizelinear", ab, l2, l1);
+        auto d4  = add_quantize_op(mm, "dequantizelinear", db, l2, l1);
+        auto q1  = add_quantize_op(mm, "quantizelinear", input, l2, l1);
+        auto d5  = add_quantize_op(mm, "dequantizelinear", q1, l2, l1);
+        auto c1  = mm.add_instruction(migraphx::make_op("convolution",
+                                                       {{"padding", {0, 0, 0, 0}},
+                                                        {"stride", {1, 1}},
+                                                        {"dilation", {1, 1}},
+                                                        {"group", 1},
+                                                        {"padding_mode", 0}}),
+                                     d5,
+                                     d1);
+        auto bc1 = mm.add_instruction(
+            migraphx::make_op("broadcast", {{"axis", 1}, {"dims", {1, 1280, 7, 7}}}), d2);
+        auto a1 = mm.add_instruction(migraphx::make_op("add"), c1, bc1);
```

Comment:
This is missing a q/dq pair for the convolution output before it goes into `add`.